aw_watcher_terminal/main.py

```diff
@@ -35,10 +38,26 @@ def main() -> None:
 
     # Start fifo listener and event_queue updater concurrently
     with ThreadPoolExecutor(max_workers=2) as executor:
-        executor.submit(run_event_queue_updater)
-        executor.submit(on_named_pipe_message,
-                        fifo_path,
-                        message_handler.handle_fifo_message)
+        futures = []
```

Comment:
Looks a little bit messy, but considering that it can handle exceptions nicely and I can't come up with a good alternative which also does that this seems pretty good.

Reply:
I could also rewrite it not to use threads, but a single loop:

```python
"""
Periodically read pipe for new messages
and call callback if one was found
"""
pipe_fd = os.open(fifo_path, os.O_RDONLY | os.O_NONBLOCK)
with os.fdopen(pipe_fd) as pipe:
    config.logger.info("Listening to pipe: {}".format(pipe_path))
    while True:
        # Check for new messages
        try:
            message = pipe.read()
        if message:
            message_handler.handle_fifo_message(message)
        except Exception as e:
            config.logger.error(e)
            traceback.print_exc()
        
        # Update Event queue
        try:
            message_handler.update_event_queue()
        except Exception as e:
            config.logger.error(e)
            traceback.print_exc()

        sleep(1)
```

Reply:
Well, if you don't need a thread this is much better